phpBB/phpbb/user.php

```diff
@@ -645,6 +637,36 @@ function format_date($gmepoch, $format = false, $forcedate = false)
 		}
 	}
 
+	/**
+	 * Create a DateTimeZone object in the context of the current user
+	 *
+	 * @param string $user_timezone Time zone of the current user.
+	 * @return DateTimeZone DateTimeZone object linked to the current users locale
+	 */
+	public function create_timezone($user_timezone = null)
+	{
+		if (!isset($this->timezone))
```

Comment:
I know this works, but I am not a fan of checking if a variable is `null` with `isset()`.
`isset()` should be used to check if a variable is defined, not if it is `null`. Atleast in my eyes.
This contradicts with the actual function name, cause it is set, it just does not have the desired value.
Personally I would use `if ($this->timezone === null)`.

Reply:
> This contradicts with the actual function name, cause it is set

Where?

Reply:
This is actually ok.

"Determine if a variable is considered set, this means if a variable is declared and is different than NULL."
- https://www.php.net/manual/en/function.isset.php

Reply:
https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/user.php#L35
As I said, I understand this will work as it currently is, but I am not a fan of using `isset()` to check if a variable is `null`, rather than if the variable actually exists.
```php
if ($something)
{
    $variable = 'i am set';
}

$var = isset($variable) ?: 'is set' : 'is not set';
```
In such a manner `isset()` makes sense to me, as the variable might be completely not defined, not initialised.
But that is not the case for `$timezone`, it is initialised, just with a value of `null`.

But as I mentioned, this might just be my view on how `isset()` should be used.

Reply:
Simply `!$this->timezone` can be used instead.

Reply:
That's exactly what I am saying @dark-1 .. I know it is correct, but that doesn't mean we should use it that way. That's the entire discussion I am bringing up.
I am indeed more in favor of `$this->timezone === null` or `!$this->timezone` for that matter.

Reply:
> Simply `!$this->timezone` can be used instead.

Nice... 👍 

---

> That's exactly what I am saying @dark-1 .. I know it is correct, but that doesn't mean we should use it that way. ...

No worries 👍